core/services/chainlink/application.go

```diff
@@ -923,6 +924,20 @@ func newCREServices(
 
 				engineRegistry := syncer.NewEngineRegistry()
 
+				globalLogger.Debugw("Creating WorkflowRegistrySyncer")
+				wfRegRid := capCfg.WorkflowRegistry().RelayID()
+				wfRegRelayer, err := relayerChainInterops.Get(wfRegRid)
+				if err != nil {
+					return nil, fmt.Errorf("could not fetch relayer %s configured for workflow registry: %w", rid, err)
+				}
+				crFactory := func(ctx context.Context, bytes []byte) (commontypes.ContractReader, error) {
+					return wfRegRelayer.NewContractReader(ctx, bytes)
+				}
+				wrVersion, err := versioning.VerifyTypeAndVersion(ctx, capCfg.WorkflowRegistry().Address(), crFactory, versioning.ContractType("WorkflowRegistry"))
```

Comment:
Could this lead to a race condition due to the time it takes to set up the contract reader? I think you might need to retry as it could fail initially

Reply:
I did add retries.. but ended up hitting a blocker in that at this point in application.go the relayer service hasn't been started. This worked back when the version switching happened in `WorkflowRegistrySyncer.Start()`, but no longer after splitting into a `v2` package.

To get around this I just added some new TOML config to manually pass the version. Definitely not as nice to use... but if we want the versions separated it's how to avoid some larger refactor in `application.go`.